pkg/cloudprovider/providers/oci/fault_domain_controller.go

```diff
@@ -0,0 +1,149 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"go.uber.org/zap"
+	"k8s.io/client-go/util/workqueue"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/oracle/oci-go-sdk/core"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+)
+
+var UpdateNodeSpecBackoff = wait.Backoff{
+	Steps:    20,
+	Duration: 50 * time.Millisecond,
+	Jitter:   1.0,
+}
+
+type FaultDomainController struct {
+	nodeInformer coreinformers.NodeInformer
+	kubeClient clientset.Interface
+	recorder record.EventRecorder
+	cloud *CloudProvider
+	queue workqueue.RateLimitingInterface
+	logger *zap.Logger
+}
+
+// NewFaultDomainController creates a FaultDomainController object
+func NewFaultDomainController(
+	nodeInformer coreinformers.NodeInformer,
+	kubeClient clientset.Interface,
+	cloud *CloudProvider) *FaultDomainController {
+
+	logger := zap.L()
```

Comment:
pass the logger from caller instead of creating a new one.

Reply:
Just saw that the *CloudProvider object contains a logger, I will use that